sdks/python/apache_beam/yaml/main.py

```diff
@@ -67,6 +105,12 @@ def _pipeline_spec_from_args(known_args):
 def run(argv=None):
   known_args, pipeline_args = _configure_parser(argv)
   pipeline_yaml = _pipeline_spec_from_args(known_args)
+  jinja_variables = _extract_jinja_variables(known_args, pipeline_args)
+  if (known_args.jinja_variables is not None or
+      known_args.flags_as_jinja_variables):
```

Comment:
```suggestion
  if len(jinja_variables) > 0:
```

We can probably simplify here.

Reply:
I want to be able to say --jinja_variables={} to enable jinja processing even if no variables are used.

Reply:
Actually, should this be in an `if` block at all? Don't we want to be able to do pre-processing even if there aren't specific variables to override (to allow things like for loops)?

Reply:
Oh, race condition here - do we want this disabled by default? I can't really imagine a scenario where you've got jinja syntax but don't want it replaced

Reply:
I was imagining that jinja templating would be the exceptional, not normal, case, and specifying variables would be the way to trigger is. But I suppose the case could be made that it's (very likely) the identity function for "plain vanilla" yaml, and if not we should perhaps resolve the ambiguity by running the preprocessor and forcing the user to escape if they want literal `{%`, etc. anywhere.

Reply:
> But I suppose the case could be made that it's (very likely) the identity function for "plain vanilla" yaml, and if not we should perhaps resolve the ambiguity by running the preprocessor and forcing the user to escape if they want literal {%, etc. anywhere.

Yeah, this is what I was thinking. This also avoids the unpleasantness of adding `jinja_variables` to your pipeline invocation changing things that are totally unrelated to those variables

Reply:
OK, we're just invoking this unconditionally now.